src/index.ts

```diff
@@ -69,6 +73,16 @@ export const useForm = <TState, TValidatedState = TState>(
     }
   }
 
+  watch(
+    state,
+    async () => {
+      if (validateOn === "change") {
+        validate()
+      }
+    },
+    { deep: true }
+  )
```

Comment:
@tommasogangemi as `validateOn` is a constant at runtime, woudn't be possible to invert the `if` and the `watch`?

```suggestion
if (validateOn === "change") {
  watch(
    state,
    validate,
    { deep: true }
  )
 }
```